### PR TITLE
Using transport helper for calling http.request().

### PR DESCRIPTION
**Pull request by dhermes**
This (reference to `http.request`) assumes (for now) that `http` is an instance of `httplib.Http`.

Alternative approach to #583. Towards #128.

Note that this PR stops short of
- just using `http` instead of `http.request`
- changing the input argument in all helpers from `http_request` to `http`

### CODE REVIEW

File: tests/contrib/test_metadata.py [hunk 1]
```diff
@@ -28,13 +28,23 @@
...
 
 
 def request_mock(status, content_type, content):
     response = http_mock.ResponseMock(
         {'status': status, 'content-type': content_type})
-    return mock.Mock(return_value=(response, content.encode('utf-8')))
+    request_method = mock.Mock(
+        return_value=(response, content.encode('utf-8')))
+    # Make sure the mock doesn't have a request attr.
+    del request_method.request
```
Comment (dhermes) [inline tests/contrib/test_metadata.py:46]: Here and elsewhere that I have to `del foo.request`, I'd rather just update the code to properly use our `http_mock.HttpMock` test class.

**Comment by jonparrott** [comment]
It still seems strange to me that `http` is an argument to `transport.request`.

**Comment by dhermes** [comment]
@jonparrott Baby steps

**Comment by jonparrott** [comment]
> @jonparrott Baby steps

But it makes me feel icky.

But more seriously this looks fine so long as there's another PR to nix the http arg.

**Comment by jonparrott** [comment]
Actually, why not go ahead and make http and optional arg to `transport.request`?

**Comment by dhermes** [comment]
I'd rather have a discussion first about your idea of "under the covers" vs. up-front `http`. Let's hold off on making it optional until we can hash out more details.

`transport` is essentially complete, except for possibly adding a helper to get status or other headers from a response object.

File: tests/test_client.py [hunk 2]
```diff
@@ -1381,6 +1390,10 @@ def _do_retrieve_scopes_test_helper(self, response, content,
         assertUrisEqual(self, token_uri, called_uri)
         logger.info.assert_called_once_with('Refreshing scopes')
 
+        http_request.assert_called_once_with(
+            token_uri, method='GET', body=None, headers=None,
+            redirections=5, connection_type=None)
```
Comment (dhermes) [inline tests/test_client.py]: @jonparrott I (think I) now see why you didn't make this assertion, the `token_uri` and `called_uri` might not be literally equal due to dictionary order in Python (flaky failed tests).

Fail by me for not noticing above

``` python
self.assertEqual(len(http_request.mock_calls), 1)
scopes_call = http_request.mock_calls[0]
```

**Comment by jonparrott** [comment]
> I'd rather have a discussion first about your idea of "under the covers" vs. up-front http. Let's hold off on making it optional until we can hash out more details.

SG.

**Comment by dhermes** [comment]
@jonparrott Good to merge? @nathanielmanistaatgoogle any thoughts?

**Comment by jonparrott** [comment]
I'm fine with it for now.

**Comment by nathanielmanistaatgoogle** [comment]
Resolve conflicts and poke me again?

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle resolved

**Pull request merged**